src/ManifoldsBase.jl

```diff
@@ -506,19 +535,23 @@ Currently the following are supported
 Keyword arguments can be used to specify tolerances.
 """
 function isapprox(M::AbstractManifold, p, q; error::Symbol = :none, kwargs...)
-    ma = check_approx(M, p, q; kwargs...)
-    if ma !== nothing
-        (error === :error) && throw(ma)
-        if isnan(ma.val)
-            s = "$(typeof(ma))\n$(ma.msg)"
-        else
-            s = "$(typeof(ma)) with $(ma.val)\n$(ma.msg)"
+    if error === :none
+        return isapprox(p, q; kwargs...)
```

Comment:
I do not like this. Especially for posts on manifolds that require information from the manifold still (I do not have an example in mind but well)
or that like FixedRankMatrices have a very special form – I would still expect them to implement the check? Or is this intended to be the new switch of logic?

Reply:
This is kind of new and old logic. New, because `isapprox` can't call `check_approx` by default (it's quite slow). Old, because right now all manifolds implement `isapprox` anyway instead of `check_approx` so there isn't really much to change.

Reply:
I'm definitely open to having a different interface but one constraint is that we can't create any error objects in the fast/default case.

Reply:
I though about this a bit yesterday evening. What I do not like in the following form is

`check_approx(M,p,q; kwargs...)` does the same as `isapprox(p,q; kwargs...)` in these lines, but the first allocates a string which we want to get rid of.
Note that the first has much more information (M, for tangent vectors even M and p) so that seems for „doing the same” a bit strange.

I am not the biggest fan of doubling logix here, but maybe that is needed, since one works on providing more information in the string along the way.

a) My idea would be an internak function that has the same signature as `check_approx` but just returns true/false, maybe `_isapprox(M,p,q;kwargs...)`?

we could also
b) have two signatures of `check_approx` something like `check_approx(M, p, q, r; kwargs...)`where `r` is something to dispatch on determining the return type? So maybe even `r=String` and `r=Bool`? The disadvanbtage of this is, it would be quite breaking and all current checks need a new `r::Type{String}`. On the other hand this might even allow for other check types later as well (no clue which ;))

Reply:
I think I prefer the `_isapprox` solution so I will go with that if you are OK with it. `r::Type{String}` seems like too much complexity added to a relatively simple thing.

Reply:
ok, I felt like the type approach would be non-breaking and easier to adapt in Manifolds, but I do not have a strong preference.

Reply:
There is not much work either way actually :slightly_smiling_face: .

Reply:
I've added `_isapprox`. It's not ideal because our traits cause type instability when using `isapprox` but I think we have to accept that.

Reply:
does the other idea (the dispatch in check) avoid that or does that have the same issue?

Reply:
I've just spent some time digging into it and the issue is that Julia really dislikes optimizing through our trait mechanism (see this if you want to dig why: https://github.com/JuliaLang/julia/blob/cd955e15fa6f55cd24a7b759d64b4ba72595e1bb/base/compiler/typelimits.jl#L17 ). It's not a problem if we don't care too much about performance or use traits just a little but in the case I've been checking (`isapprox(Euclidean(2), [2, 3], [3, 4])`) it is quite complicated.

Reply:
Ah, I've found a solution, though it's quite ugly. I will post here later about it.

Reply:
😱ugly code!

We will see, maybe we find something nicer in the end

Reply:
It seems to work, and the ugly part looks like that:
```julia
for trait_type in [
    TraitList{<:IsDefaultMetric},
    TraitList{<:IsDefaultConnection},
    TraitList{IsMetricManifold},
    TraitList{IsConnectionManifold},
]
    @eval begin
        ManifoldsBase.@next_trait_function $trait_type isapprox(
            M::AbstractDecoratorManifold,
            p,
            q;
            kwargs...,
        )
        ManifoldsBase.@next_trait_function $trait_type isapprox(
            M::AbstractDecoratorManifold,
            p,
            X,
            Y;
            kwargs...,
        )
    end
end
```
But we may need that for more functions and more traits, maybe even all of them.

Reply:
Which basically skips certain traits? Isn't it better to make is approx transparent to these?

Reply:
No, it's functionally equivalent to what we have now. It just makes more "copies" of the go-to-the-next-trait method so that we don't have method recursion.

What do you mean by making approx transparent to these? Not adding these traits to `isapprox` dispatch?

Reply:
Well preeviously some trait not doing anything on a function is what we called them “transparent”, right? And I thought what you are introducing there was actually the default behaviour already? But I might have to check.

Reply:
I just remember that “pass through for default” was not so complicated as in your code.

Reply:
Ah, I too often think in the “just work” regime of code. So my intuition here was correct, just not my compiler ideas. But well, if this is necessary then lets leave it at that.